cnd/src/swap_protocols/rfc003/bob/mod.rs

```diff
@@ -21,9 +24,9 @@ pub struct State<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset> {
     pub swap_communication: SwapCommunication<AL, BL, AA, BA>,
     pub alpha_ledger_state: LedgerState<AL>,
     pub beta_ledger_state: LedgerState<BL>,
-    #[derivative(Debug = "ignore")]
-    pub secret_source: Arc<dyn SecretSource>,
-    pub secret: Option<Secret>,
+    #[derivative(Debug = "ignore", PartialEq = "ignore")]
+    pub secret_source: Seed, // Used to derive identities for refund/redeem.
```

Comment:
If you want to add comments, please use rustdoc. (triple slash comment in the line above the field.

Reply:
This is not, in my opinion, a comment for consumers of this module so it doesn't get a rustdoc.  Its a comment for internal usage of this module because the variable name/type needs further explanation.  If my understanding of use of rustdoc vs `//` is wrong I'm happy to be schooled.  Also, I think that if it does require a rustdoc comment then all of the fields here should have one - if this is the case though, we probably should have a lot more documentation.

Reply:
> we probably should have a lot more documentation.

Yes!

>  Also, I think that if it does require a rustdoc comment then all of the fields here should have one

Yes!

>  If my understanding of use of rustdoc vs // is wrong I'm happy to be schooled.

I wouldn't make a distinction here. rustdoc is Rust's way of adding documentation to the source code. Because it uses a well-defined format, it can then automatically generate documentation. IMO, there is no downside to using it, so I would just opt for rustdoc :)

Reply:
> I wouldn't make a distinction here. rustdoc is Rust's way of adding documentation to the source code. Because it uses a well-defined format, it can then automatically generate documentation. IMO, there is no downside to using it, so I would just opt for rustdoc :)

I have to disagree with you here @thomaseizinger, rustdoc is for comments that should be included in the rendered documentation _not_ for random code comments (that arguably should not need to exist in the first place).  I definitely feel there is a usecase for `//` comments.  Whether this case in question is one of those is open to debate.

Reply:
>  I definitely feel there is a usecase for // comments. Whether this case in question is one of those is open to debate.

I think having such comments can be worthwhile if they are inline with the source-code.
You are adding documentation to a struct field here, I'd argue that is what rustdoc is for (among other things).